packages/react-devtools-shared/src/backend/renderer.js

```diff
@@ -1272,6 +1275,13 @@ export function attach(
             }
           }
         }
+        return [legacyContext, modernContext];
+      case ElementTypeFunction:
+        const dependencies = fiber.dependencies;
+        if (dependencies && dependencies.firstContext) {
+          modernContext = dependencies.firstContext.memoizedValue;
+        }
```

Comment:
Huh. I guess this _works_ although TBH I didn't expect it to when I first read it 😅 I guess we always recreate the `firstContext` object even if a subsequent context is updated, b'c of the `next` pointer. Cool 👍🏼

Reply:
Lol I felt the same way. Originally I planned on traversing each `next` child, but was testing different cases with `Object.is` and discovered that it worked even if I didn't check every context. So just went with it 😅

Reply:
Well, looks like I was wrong. This only worked in my manual test scenarios because the `memoizedValue` was an object (which was recreated). However I totally missed that `memoizedValue` can be a primitive, as it is in the test cases.

So I fixed up the logic to traverse the context list and compare each `memoizedValue` - https://github.com/facebook/react/pull/22047/commits/c540f2a1a2e2df03b816844a7937b7100c357b83

Reply:
Also updated the test case & snapshots to validate consuming multiple contexts - https://github.com/facebook/react/pull/22047/commits/c540f2a1a2e2df03b816844a7937b7100c357b83

Reply:
> Well, looks like I was wrong. This only worked in my manual test scenarios because the `memoizedValue` was an object (which was recreated). However I totally missed that `memoizedValue` can be a primitive, as it is in the test cases.

Okay that's kind of funny 😃